worlds/aquaria/Regions.py

```diff
@@ -1152,79 +1152,79 @@ def __adjusting_manual_rules(self) -> None:
     def __no_progression_hard_or_hidden_location(self) -> None:
         self.multiworld.get_location("Energy Temple boss area, Fallen God Tooth",
                                      self.player).item_rule = \
-            lambda item: item.classification != ItemClassification.progression
+            lambda item: not item.advancement
```

Comment:
For future reference, each lambda lives in memory, so it would be more efficient to define this once and paste it across all relevant locations.

Reply:
Do you mean that lambdas are never collected? Like for the rest of the execution of the program?